### PR TITLE
Allow setting the soldiers of a tower

### LINKED ISSUE #415: Towers should search for all required soldier types simulatneously

### PR DESCRIPTION
**Pull request by andreas-eberle**
fixes #415, #381

### CODE REVIEW

**Comment by andreas-eberle** [comment]
@codingberlin: One of the AI tests is failing. I guess this is because the towers are not completely filled with soldiers by default. Can you update the AI to set the towers to be fully occupied?

**Comment by andreas-eberle** [comment]
@michaelzangl: Can you have a look at the changes?

File: jsettlers.common/src/main/java/jsettlers/common/utils/collections/map/OrderedMap.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by Andreas Eberle on 06.05.2016.
+ */
+public class OrderedMap<K, V> implements Serializable {
```
Comment (michaelzangl) [inline jsettlers.common/src/main/java/jsettlers/common/utils/collections/map/OrderedMap.java]: What is the difference to LinkedHashSet?

**Comment by michaelzangl** [comment]
The rest seems fine to me.

File: jsettlers.common/src/main/java/jsettlers/common/utils/collections/map/OrderedMap.java (hunk 1 above)
Comment (andreas-eberle) [inline jsettlers.common/src/main/java/jsettlers/common/utils/collections/map/OrderedMap.java]: The problem with LinkedHashMap is that it doesn't serialize consistently. If you serialize, deserialize and serialize a LinkedHashMap again, the bytes of the two serializations can be different. However, we need them to be the same to be able to see if savegames differ due to actual changes.

**Pull request merged**